src/DepthImageToLaserScanROS.cpp

```diff
@@ -31,61 +31,78 @@
  * Author: Chad Rockey
  */
 
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/msg/camera_info.hpp>
+#include <sensor_msgs/msg/image.hpp>
+#include <sensor_msgs/msg/laser_scan.hpp>
+
+#include <functional>
+
 #include <depthimage_to_laserscan/DepthImageToLaserScanROS.h>
 
+#define ROS_ERROR printf
+
 using namespace depthimage_to_laserscan;
-  
-DepthImageToLaserScanROS::DepthImageToLaserScanROS(ros::NodeHandle& n, ros::NodeHandle& pnh):pnh_(pnh), it_(n), srv_(pnh) {
-  boost::mutex::scoped_lock lock(connect_mutex_);
-  
-  // Dynamic Reconfigure
-  dynamic_reconfigure::Server<depthimage_to_laserscan::DepthConfig>::CallbackType f;
-  f = boost::bind(&DepthImageToLaserScanROS::reconfigureCb, this, _1, _2);
-  srv_.setCallback(f);
-  
-  // Lazy subscription to depth image topic
-  pub_ = n.advertise<sensor_msgs::LaserScan>("scan", 10, boost::bind(&DepthImageToLaserScanROS::connectCb, this, _1), boost::bind(&DepthImageToLaserScanROS::disconnectCb, this, _1));
+
+DepthImageToLaserScanROS::DepthImageToLaserScanROS(rclcpp::node::Node::SharedPtr & node):node_(node) {
```

Comment:
This is a pattern I've seen in some other ROS nodes, and upon reflection, I thought it was a good idea.  The reason is to keep a reference to the node in-scope.  Imagine you have the following (pseudo-)code:

```
void init()
{
  rclcpp::node::Node::SharedPtr node = rclcpp::node::Node::make_shared("depthimage_to_laserscan");

  depthimage_to_laserscan::DepthImageToLaserScanROS dtl(node);
}

int main()
{
    init();
}
```

In this case, when init() completes, node goes out of scope and is destroyed.  While this doesn't quite matter right now (the DepthImageToLaserScanROS class never references node outside of the constructor), it seems to me that it is a good idea to hold onto a reference since we may need it in the future.